### ISSUE TITLE
org.openqa.selenium.WebDriverException: Socket timeout reading Marionette handshake data: Resource temporarily unavailable (os error 11)

### ISSUE THREAD

**Issue by weiding111** [mozilla/geckodriver#1470]
## System

* Version: geckodriver-v0.23.0-linux64
* Platform: LINUX  version 4.1.12-124.17.2.el7uek.x86_64 (gcc version 4.8.5 20150623 (Red Hat 4.8.5-16.0.3) (GCC) )
* Firefox: firefox 61.0
* Selenium: browser: firefox; selenium version: 3.13.0

## Testcase

This issue is intermittent and appears when some test is failed. Then following tests are all failed due to this issue. From the browser, the phenomenon  is that the browser launched and then tear down.

## Stacktrace

org.openqa.selenium.WebDriverException: Socket timeout reading Marionette handshake data: Resource temporarily unavailable (os error 11)

**Comment by andreastt** [comment]
Could you please attach a [trace-level log] from geckodriver? Read more about reporting actionable bugs in our [contribution guidelines].

[trace-level log]: https://firefox-source-docs.mozilla.org/testing/geckodriver/geckodriver/TraceLogs.html
[contribution guidelines]: https://firefox-source-docs.mozilla.org/testing/geckodriver/geckodriver/Bugs.html

**Comment by barancev** [comment]
We see this failure on travis too, for example
https://travis-ci.org/SeleniumHQ/selenium/jobs/481828240
https://travis-ci.org/SeleniumHQ/selenium/jobs/480205363
https://travis-ci.org/SeleniumHQ/selenium/jobs/478786196

**Comment by whimboo** [comment]
@barancev the problem here is always that Marionette is not able to bind to the port as selected by geckodriver:

> 1547925593150	Marionette	FATAL	Remote protocol server failed to start: Error: Could not bind to port 46112 (NS_ERROR_SOCKET_ADDRESS_IN_USE)(chrome://marionette/content/server.js:87:17) JS Stack trace: set acceptConnections@server.js:87:17

I wonder if this could be somewhat related to https://github.com/mozilla/geckodriver/issues/979#issuecomment-417321586.

**Comment by andreastt** [comment]
Maybe we should shut down Firefox if `./firefox --marionette` fails to bind to the port.

**Comment by whimboo** [comment]
We do this already, but it wouldn't solve the underlying problem that the port Marionette tries to connect to is most likeley still hold by geckodriver.

**Comment by thomasbecker** [comment]
I can confirm this error. We just upgraded from ff59 and gecko 0.20.1. Never seen this issue before and it occurs regularly (about every 2nd to 3rd build) since the update.

Let me know if I can help by providing further information. I can also try some more ff/gecko versions to pinpoint the version that introduced this behaviour.

**Comment by whimboo** [comment]
Thank you @thomasbecker. If you could try geckodriver releases 0.21.0 up to 0.23.0 that would be great. If all of those are failing, it would be good to bisect Firefox releases. But I assume this might be geckodriver related. Thanks!

**Comment by thomasbecker** [comment]
@whimboo Sorry for the delay. Just didn't had the time yet.

I've reproduced the issue with 0.23.0. So it definetly occurs with gecko 0.23.0 and ff65.0.1. I couldn't reproduce it with 0.22.0 and 0.21.0. But here's the catch...the error occurs intermittently and failing random tests with 0.23.0. I did a couple of runs with 0.22.0 and 0.21.0, but that's not a guarantee it's really not in versions < 0.23.0.

Bottomline:  It's very likely that 0.22.0 doesn't have the error and it's sure that the error occurs with 0.23.0.

I will try to run some more tests with 0.22.0 on monday. For now I've to roll it back as it seems to produce some other random errors which I will have to debug first.

**Comment by whimboo** [comment]
At least nothing in the 0.23.0 release stands out for me to be causing this problem:
https://github.com/mozilla/geckodriver/releases/tag/v0.23.0

But please continue and let me know. Thanks a lot!

**Comment by thomasbecker** [comment]
I did some more runs with 0.22.0 and as already stated we're getting some random test errors which I need to investigate. So I can't stay on 0.22.0 for now. But this issue's error didn't appear in any of the builds. So it really seems that it was somehow introduced with 0.23.0.

**Comment by whimboo** [comment]
@thomasbecker btw how does your test scenario exactly look like? Are you running multiple tests in parallel? Maybe you an give more details. Thanks.

**Comment by thomasbecker** [comment]
@whimboo Yes, I can. But it's complicated. :)

We've got about a thousand Selenium Tests which take more than 3h to run sequentially. We had no luck running them in parallel using JUnit features (we get random intermittent test failures doing so), but we need builds finished in about 10 minutes.

So we split them up into backpacks of tests. We average each tests duration from the last couple of builds and that way we get backpacks of about the same (duration) size.
These backpacks of test classes are then run inside individual docker containers using maven. The containers contain firefox, gecko and the running application we're testing. 
Each container starts only a single webdriver which is then shared across all tests running inside that backpack/container.

Currently we're running 33 containers in parallel and all tests finish within 10 minutes.

I didn't test this extensive, but it seems to take about 2-4 runs of all tests to get the issue described here once. I've done about 10 runs with 0.22.0 and the issue didn't occur whereas I've been able to reproduce it with 0.23.0 a couple of times. 

If it helps I could enable some debug output in firefox/gecko and try to reproduce this again. Then let me know which debug options you want to have enabled and I will try to get some logs for you.

**Comment by sanduni1992** [comment]
@thomasbecker I’m also getting this error and I’m looking for solution for this. If this issue fixed in your automation suit can you give details of your firefox version, gecko driver version & selenium version. This will be be very helpful to stop the intermittent errors in my automation suit.

**Comment by dantamont** [comment]
Any updates on this?  I'm getting the same error using FireFox 65.0.1 and version 0.24 of gecko driver:

Stack trace: 

[Fri Mar 22 06:41:49.805426 2019] [wsgi:error] [pid 21077:tid 140067426567936] [remote 172.114.238.101:62454]   File "/root/seeker/senv/lib/python3.7/site-packages/selenium/webdriver/firefox/webdriver.py", line 167, in __init__

[Fri Mar 22 06:41:49.805431 2019] [wsgi:error] [pid 21077:tid 140067426567936] [remote 172.114.238.101:62454]     keep_alive=True)

[Fri Mar 22 06:41:49.805436 2019] [wsgi:error] [pid 21077:tid 140067426567936] [remote 172.114.238.101:62454]   File "/root/seeker/senv/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 156, in __init__

[Fri Mar 22 06:41:49.805441 2019] [wsgi:error] [pid 21077:tid 140067426567936] [remote 172.114.238.101:62454]     self.start_session(capabilities, browser_profile)

[Fri Mar 22 06:41:49.805447 2019] [wsgi:error] [pid 21077:tid 140067426567936] [remote 172.114.238.101:62454]   File "/root/seeker/senv/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 251, in start_session

[Fri Mar 22 06:41:49.805452 2019] [wsgi:error] [pid 21077:tid 140067426567936] [remote 172.114.238.101:62454]     response = self.execute(Command.NEW_SESSION, parameters)

[Fri Mar 22 06:41:49.805457 2019] [wsgi:error] [pid 21077:tid 140067426567936] [remote 172.114.238.101:62454]   File "/root/seeker/senv/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 320, in execute

[Fri Mar 22 06:41:49.805462 2019] [wsgi:error] [pid 21077:tid 140067426567936] [remote 172.114.238.101:62454]     self.error_handler.check_response(response)

[Fri Mar 22 06:41:49.805468 2019] [wsgi:error] [pid 21077:tid 140067426567936] [remote 172.114.238.101:62454]   File "/root/seeker/senv/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response

[Fri Mar 22 06:41:49.805473 2019] [wsgi:error] [pid 21077:tid 140067426567936] [remote 172.114.238.101:62454]     raise exception_class(message, screen, stacktrace)

[Fri Mar 22 06:41:49.805478 2019] [wsgi:error] [pid 21077:tid 140067426567936] [remote 172.114.238.101:62454] selenium.common.exceptions.WebDriverException: Message: Socket timeout reading Marionette handshake data: Resource temporarily unavailable (os error 11)

**Comment by thomasbecker** [comment]
@sanduni1992 Sorry for the late reply. Somehow missed your message. We don't seem to have this issue with gecko 0.22.0. If you have the exact same error as described in this issue here with 0.22.0 I'd guess that @whimboo would be interested in some more details.

**Comment by sanduni1992** [comment]
@thomasbecker , @whimboo  with the gecko driver 0.22.0 I got the above issue intermittently. so I decided to downgrade to FF59 & gecko driver 0.20.1 and afterwards I didn't face that issue. But I'm looking forward to upgrade it with latest compatible gecko driver & firefox. please let me know if there are any firefox version & gecko version which compatible with each other. Thank you !!

**Comment by flickerlight** [comment]
Hi, Any updates for this issue? I'm getting this error, too.

Version: geckodriver-v0.24.0-linux64
Platform: ubuntu 16.04.5
Firefox: firefox 68.0b11

org.openqa.selenium.WebDriverException: Socket timeout reading Marionette handshake data: Resource temporarily unavailable (os error 11)
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'kinetic', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.4.0-131-generic', java.version: '1.8.0_212'
Driver info: driver.version: FirefoxDriver
remote stacktrace: 

sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
java.lang.reflect.Constructor.newInstance(Constructor.java:423)
org.openqa.selenium.remote.W3CHandshakeResponse.lambda$errorHandler$0(W3CHandshakeResponse.java:62)
org.openqa.selenium.remote.HandshakeResponse.lambda$getResponseFunction$0(HandshakeResponse.java:30)
org.openqa.selenium.remote.ProtocolHandshake.lambda$createSession$0(ProtocolHandshake.java:126)
java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:499)
java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:486)
java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:152)
java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:464)
org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:128)
org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:74)
org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:136)
org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:83)
org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:552)
org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:213)
org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:131)
org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:147)
com.cisco.sft.gauge_test.webUITestSteps.openBrowser(webUITestSteps.java:85)
sun.reflect.GeneratedMethodAccessor7.invoke(Unknown Source)
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
java.lang.reflect.Method.invoke(Method.java:498)
com.thoughtworks.gauge.execution.MethodExecutor.execute(MethodExecutor.java:38)
com.thoughtworks.gauge.execution.StepExecutionStage.executeStepMethod(StepExecutionStage.java:76)
com.thoughtworks.gauge.execution.StepExecutionStage.executeStep(StepExecutionStage.java:69)
com.thoughtworks.gauge.execution.StepExecutionStage.execute(StepExecutionStage.java:51)
com.thoughtworks.gauge.execution.AbstractExecutionStage.executeNext(AbstractExecutionStage.java:24)
com.thoughtworks.gauge.execution.HookExecutionStage.execute(HookExecutionStage.java:43)
com.thoughtworks.gauge.execution.ExecutionPipeline.start(ExecutionPipeline.java:29)
com.thoughtworks.gauge.processor.ExecuteStepProcessor.process(ExecuteStepProcessor.java:48)
com.thoughtworks.gauge.connection.MessageDispatcher.dispatchMessages(MessageDispatcher.java:83)
com.thoughtworks.gauge.GaugeRuntime.dispatchMessages(GaugeRuntime.java:127)
com.thoughtworks.gauge.GaugeRuntime.lambda$connectSynchronously$1(GaugeRuntime.java:107)
java.lang.Thread.run(Thread.java:748)

**Comment by Rodaxfck** [comment]
Hello,

Version: geckodriver 0.24.0 ( 2019-01-28)
Platform: Ubuntu 14.04 (on Jenkins slave)
Firefox: 66.0.3

| INFO | Opening browser 'Firefox' to base url 'https://preprod.m2m.orange.com/'.
| FAIL | WebDriverException: Message: Socket timeout reading Marionette handshake data: Resource temporarily unavailable (os error 11)

I have the same error !
Randomly, and not in a specific test case...
I have around 3-4 failed test cases with this error / 150 tests in a test suite.
But only sometimes, I can have 0 failed because of this error. It's totally random.

My test cases are coded/executed with Robot Framework/SeleniumLibrary, and to open the browser, I use the keyword "Open Browser".
I have noticed that if I use the keyword "Create Webdriver", I don't have the error (and it's the only one keyword I can use to enable TRACE logs in geckodriver.log :'( I can't do it with "Open Browser" keyword) ! 
[Code of "Open Browser" and "Create Webdriver" heve](https://github.com/robotframework/SeleniumLibrary/blob/master/src/SeleniumLibrary/keywords/browsermanagement.py)

**Comment by Rodaxfck** [comment]
I have two test suites runned in parallel so this is the mess in logs but in geckodriver.log I've found this (One "Killing Process" with the error):

1563422824342	mozrunner::runner	DEBUG	Killing process 29390
1563422824342	webdriver::server	DEBUG	<- 500 Internal Server Error {"value":{"error":"unknown error","message":"Socket timeout reading Marionette handshake data: Resource temporarily unavailable (os error 11)","stacktrace":""}}

I think the error appears after a "Close Browser"

**Comment by feal94** [comment]
Hello, I run into this same error when running with geckodriver 0.24.0  and Firefox 68.0.1. 
I'm running my script with joblib.Parallel and 20 jobs. If I use only one job everything runs smoothly. 

Traceback: 

```
File "/home/alvarofeal/.local/lib/python3.5/site-packages/seleniumwire/webdriver/browser.py", line 46, in __init__
    super().__init__(*args, **kwargs)
  File "/home/alvarofeal/.local/lib/python3.5/site-packages/selenium/webdriver/firefox/webdriver.py", line 174, in __init__
    keep_alive=True)
  File "/home/alvarofeal/.local/lib/python3.5/site-packages/selenium/webdriver/remote/webdriver.py", line 157, in __init__
    self.start_session(capabilities, browser_profile)
  File "/home/alvarofeal/.local/lib/python3.5/site-packages/selenium/webdriver/remote/webdriver.py", line 252, in start_session
    response = self.execute(Command.NEW_SESSION, parameters)
  File "/home/alvarofeal/.local/lib/python3.5/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/home/alvarofeal/.local/lib/python3.5/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: Socket timeout reading Marionette handshake data: Resource temporarily unavailable (os error 11)
```

And tail of geckodriver.log: 

```
1564404864301	Marionette	INFO	Stopped listening on port 32959

###!!! [Child][MessageChannel] Error: (msgtype=0x53000D,name=PHttpChannel::Msg_RemoveCorsPreflightCacheEntry) Closed channel: cannot send/recv

1564404870033	Marionette	INFO	Stopped listening on port 36463
[Child 267179, Chrome_ChildThread] WARNING: pipe error: Broken pipe: file /builds/worker/workspace/build/src/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 727
[Child 267179, Chrome_ChildThread] WARNING: pipe error: Broken pipe: file /builds/worker/workspace/build/src/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 727
[Child 267179, Chrome_ChildThread] WARNING: pipe error: Broken pipe: file /builds/worker/workspace/build/src/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 727
[Child 267179, Chrome_ChildThread] WARNING: pipe error: Broken pipe: file /builds/worker/workspace/build/src/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 727
[Child 267179, Chrome_ChildThread] WARNING: pipe error: Broken pipe: file /builds/worker/workspace/build/src/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 727
```

**Comment by sofiand-png** [comment]
I have the same problem when running in centOS with latest firefox and geckodriver.

**Comment by Rodaxfck** [comment]
Still no fix for this issue ?.. I have the same problem as @feal94 with ~150 parallel tests. Random errors in the middle of PASS test cases...
Updating to geckodriver 0.25.0 don't change anything

Version: geckodriver 0.25.0 (bdb64cf16b68 2019-09-10 00:45 +0300)
Platform: Ubuntu 14.04
Firefox: 66.0.3 (no upgrade available for Ubuntu 14)

**Comment by whimboo** [comment]
With the upcoming 0.26.0 release there will actually be an improvement in how we connect to Marionette. That fix has already been landed in our repository, so if you could spend some minutes and test it, I would kind appreciate your help.

To get such a build follow the steps below:

1) Go to https://treeherder.mozilla.org/#/jobs?repo=autoland&searchStr=geckodriver&revision=4a3a7ec8d813d1f319d453abbcce3d43acb908db
2) Click on one of those green `geckodriver` entries, which matches your platform (eg. for Linux64 use the one with `TL`)
3) Check the opened panel below, and download the `geckodriver.tar.xz` artifact

Please let us know if that helps you, and if that failure no longer appears.

**Comment by Rodaxfck** [comment]
Thank you very much @whimboo. I'll try it as soon as possible

**Comment by Rodaxfck** [comment]
@whimboo,
As it's completely random, I cannot confirm yet if the preview version of geckodriver has resolved my problem but... After 3 parallel executions (1h30 each), I don't have any errors anymore ! 

I will come back if I find errors but I think you resolved my issue :D 

Thank you !!
BR

**Comment by whimboo** [comment]
That sounds good! @thomasbecker if you are still seeing this problem too and could verify the fix that would be fantastic. Thanks!

**Comment by Rodaxfck** [comment]
Ok I talked too quickly :disappointed: ...

In the middle of PASS test cases, 5 test cases are FAIL with this first error : 
"WebDriverException: Message: Failed to decode response from marionette" 

Followed by some other errors : 
"InvalidSessionIdException: Message: Tried to run command without establishing a connection"

No more errors since the "Close All Browsers" keyword, others test cases after that are PASS.
Sorry for my answer @whimboo ...
BR

**Comment by whimboo** [comment]
@Rodaxfck please provide some trace logs. Those all look unrelated to the problem covered by this issue. There might have been a crash of Firefox.

**Comment by Rodaxfck** [comment]
Here is a part of my 285mb geckodriver.log file in TRACE debug mode... 
In the middle of PASS test cases, I see a word "MOZ_CRASHREPORTER_SHUTDOWN" but I don't know what is it, and also my error :

1570005285450	webdriver::server	DEBUG	<- 200 OK {"value":true}
1570005285451	webdriver::server	DEBUG	-> POST /session/fe096e1c-79f0-4ac3-9d51-2d0cf7908e3f/elements {"using": "xpath", "value": "//div[@class='col-xs-12 table-control']/button[@title='Supprimer']"}
1570005285452	Marionette	DEBUG	0 -> [0,171,"WebDriver:FindElements",{"using":"xpath","value":"//div[@class='col-xs-12 table-control']/button[@title='Supprimer']"}]
1570005285454	Marionette	DEBUG	0 <- [1,171,null,[{"element-6066-11e4-a52e-4f735466cecf":"98bffdcd-a4b2-4455-b6d5-39440a8ad0c3"}]]
1570005285454	webdriver::server	DEBUG	<- 200 OK {"value":[{"element-6066-11e4-a52e-4f735466cecf":"98bffdcd-a4b2-4455-b6d5-39440a8ad0c3"}]}
1570005285455	webdriver::server	DEBUG	-> POST /session/fe096e1c-79f0-4ac3-9d51-2d0cf7908e3f/element/98bffdcd-a4b2-4455-b6d5-39440a8ad0c3/click {"id": "98bffdcd-a4b2-4455-b6d5-39440a8ad0c3"}
1570005285455	Marionette	DEBUG	0 -> [0,172,"WebDriver:ElementClick",{"id":"98bffdcd-a4b2-4455-b6d5-39440a8ad0c3"}]
1570005285793	Marionette	DEBUG	[4294967297] Canceled page load listener because no navigation has been detected
1570005285793	Marionette	DEBUG	0 <- [1,172,null,{"value":null}]
1570005285794	webdriver::server	DEBUG	<- 200 OK {"value":null}

###!!! [Parent][MessageChannel] Error: (msgtype=0x1E0087,name=PBrowser::Msg_Destroy) Channel error: cannot send/recv

1570005290397	Marionette	TRACE	[11] Frame script loaded
1570005290397	Marionette	TRACE	[11] Frame script registered
A content process crashed and **MOZ_CRASHREPORTER_SHUTDOWN** is set, shutting down
1570005290498	Marionette	DEBUG	Closed connection 0
[Parent 3600, Gecko_IOThread] WARNING: pipe error (78): Connection reset by peer: file /build/firefox-sc3iFa/firefox-66.0.3+build1/ipc/chromium/src/chrome/common/ipc_channel_posix.cc, line 357
1570005290561	Marionette	TRACE	Received observer notification xpcom-will-shutdown
1570005290563	Marionette	DEBUG	Resetting recommended pref apz.content_response_timeout
1570005290563	Marionette	DEBUG	Resetting recommended pref browser.contentblocking.introCount
1570005290563	Marionette	DEBUG	Resetting recommended pref browser.download.panel.shown
1570005290563	Marionette	DEBUG	Resetting recommended pref browser.newtabpage.enabled
1570005290563	Marionette	DEBUG	Resetting recommended pref browser.pagethumbnails.capturing_disabled
1570005290563	Marionette	DEBUG	Resetting recommended pref browser.search.update
1570005290563	Marionette	DEBUG	Resetting recommended pref browser.tabs.disableBackgroundZombification
1570005290564	Marionette	DEBUG	Resetting recommended pref browser.tabs.warnOnCloseOtherTabs
1570005290564	Marionette	DEBUG	Resetting recommended pref browser.tabs.warnOnOpen
1570005290564	Marionette	DEBUG	Resetting recommended pref browser.usedOnWindows10.introURL
1570005290564	Marionette	DEBUG	Resetting recommended pref browser.urlbar.suggest.searches
1570005290564	Marionette	DEBUG	Resetting recommended pref datareporting.policy.dataSubmissionPolicyAccepted
1570005290564	Marionette	DEBUG	Resetting recommended pref dom.disable_beforeunload
1570005290564	Marionette	DEBUG	Resetting recommended pref dom.disable_open_during_load
1570005290565	Marionette	DEBUG	Resetting recommended pref dom.file.createInChild
1570005290565	Marionette	DEBUG	Resetting recommended pref dom.max_chrome_script_run_time
1570005290565	Marionette	DEBUG	Resetting recommended pref dom.max_script_run_time
1570005290565	Marionette	DEBUG	Resetting recommended pref extensions.getAddons.cache.enabled
1570005290565	Marionette	DEBUG	Resetting recommended pref extensions.webservice.discoverURL
1570005290565	Marionette	DEBUG	Resetting recommended pref network.http.prompt-temp-redirect
1570005290565	Marionette	DEBUG	Resetting recommended pref network.http.speculative-parallel-limit
1570005290565	Marionette	DEBUG	Resetting recommended pref security.fileuri.strict_origin_policy
1570005290565	Marionette	DEBUG	Resetting recommended pref security.notification_enable_delay
1570005290565	Marionette	DEBUG	Resetting recommended pref signon.autofillForms
1570005290565	Marionette	DEBUG	Resetting recommended pref signon.rememberSignons
1570005290566	Marionette	DEBUG	Resetting recommended pref toolkit.cosmeticAnimations.enabled
1570005290566	Marionette	INFO	Stopped listening on port 57551
1570005290566	Marionette	DEBUG	Remote service is inactive

(firefox:3600): GLib-GObject-CRITICAL **: g_object_ref: assertion 'object->ref_count > 0' failed

(firefox:3600): GLib-GObject-CRITICAL **: g_object_unref: assertion 'object->ref_count > 0' failed
1570005290798	webdriver::server	DEBUG	-> POST /session/fe096e1c-79f0-4ac3-9d51-2d0cf7908e3f/elements {"using": "css selector", "value": "[id=\"modal-confirm\"]"}
1570005290798	webdriver::server	DEBUG	Deleting session
1570005290898	geckodriver::marionette	DEBUG	Browser process stopped: exit code: 0
1570005290904	webdriver::server	DEBUG	<- 500 Internal Server Error {"value":{"error":"unknown error","message":"**Failed to decode response from marionette**","stacktrace":""}}
1570005290906	webdriver::server	DEBUG	-> GET /session/fe096e1c-79f0-4ac3-9d51-2d0cf7908e3f/screenshot 
1570005290906	webdriver::server	DEBUG	<- 404 Not Found {"value":{"error":"**invalid session id","message":"Tried to run command without establishing a connection**","stacktrace":""}}
1570005290911	webdriver::server	DEBUG	-> DELETE /session/fe096e1c-79f0-4ac3-9d51-2d0cf7908e3f 
1570005290911	webdriver::server	DEBUG	<- 404 Not Found {"value":{"error":**"invalid session id","message":"Tried to run command without establishing a connection**","stacktrace":""}}
1570005291931	mozrunner::runner	INFO	Running command: "/usr/bin/firefox" "-marionette" "-foreground" "-no-remote" "-profile" "/tmp/rust_mozprofileNsZR3s"
1570005291932	geckodriver::marionette	DEBUG	Waiting 60s to connect to browser on 127.0.0.1:48290

**Comment by whimboo** [comment]
Yes, as I have expected that indicates that a content process (tab) has been crashed. By default Firefox will shutdown because stable tests cannot be guaranteed afterward. I would kinda appreciate if you could help investigating this problem. Therefore have a look at https://firefox-source-docs.mozilla.org/testing/geckodriver/CrashReports.html. 

Also it would be great if we could move this conversation over to a new issue. @Rodaxfck mind creating one? The remaining issue for you is really off-topic here. Thanks!

**Comment by Rodaxfck** [comment]
Hi,
I tried to use a custom profile with "-profile" args to get the minidump files but I think I have now the issue https://bugzilla.mozilla.org/show_bug.cgi?id=1421766 ... 
Because firefox starts without url, is blocked and few seconds later : "TimeoutException: Message: connection refused".
So I think I will deal with all these random errors and wait for the [0.26.0](https://bugzilla.mozilla.org/show_bug.cgi?id=1573798)/[0.27.0](https://bugzilla.mozilla.org/show_bug.cgi?id=1584911) release, because these workarounds to get minidump/custom profile are too difficult for me.

Thank you anyway, you resolved the "Resource temporarily unavailable" error I think :)

**Comment by whimboo** [comment]
@weiding111, you as original reporter of this issue can you please check with the release candidates for [geckodriver 0.26.0](https://github.com/mozilla/geckodriver/releases/tag/v0.26.0) if that now works for you?

I consider this fixed but would be happy to reopen in case you still see it. If that is the case please provide trace level logs again. Thanks!
